### ISSUE TITLE
implement caching of contribution history

### ISSUE THREAD

**Issue by brenzi** [integritee-network/crowdloan#25]
Loading all the contribution history dynamically is too slow.

# option 1: manual, file-based cache

We could have a hard-coded cache of previous contributions committed to the GH repo in regular intervals manually or maybe even with a scheduled GH action

The page should then pre-populate the list of contributions from the cache and load recent contributions dynamically

# option 2: index contributions ourselves

by scanning Kusama ourselves and maintaining our own database (possibly using polkascan code?)

**Comment by brenzi** [comment]
@Alirazadar may I ask you to comment on option 1? How long would you need for this? I believe this would be the leanest solution.

This will allow to add the following features to the crowdloan page:
* show number of unique crowdlender accounts
* show leaderboard, even if >100 lends (we're already at 83 now)
* show history plot: KSM raised vs time

**Comment by brenzi** [comment]
@Alirazadar one more reason to use a proper database:
* we are thinking about implementing a referral program which means that we will have to generate and store referral keys along with contribution adresses. As this will not be stored on chain, we need a database we can dynamically write to.
* As mentioned, I could set up some mySQL dB and store the credentials in GH secrets, so you can use it from an env variable

**Comment by Alirazadar** [comment]
@brenzi yes, understandable. I will have a look at both options. Thank you for explaining!

**Comment by brenzi** [comment]
please consider #37 which may require our own database (not necessarily indexing ourselves, we could still use subscan for this, but cache the data in our own db

**Comment by brenzi** [comment]
we will first migrate to a stateful webpage with a mysql db and then come back to this
